runtime/lua/vim/ui.lua

```diff
@@ -129,7 +129,16 @@ function M.open(path)
     path = { path, 'string' },
   })
   local is_uri = path:match('%w+:')
-  if not is_uri then
+  if is_uri then
+    local help = path:match('^help://(.+)$')
```

Comment:
👍 For reference, with [on_func](https://github.com/neovim/neovim/pull/22598) (or something like it) we could potentially have buffer-local function overrides. But handling `help://` globally in `vim.ui.open` is quite fine for now.